src/Aspire.Cli/Commands/DeployCommand.cs

```diff
@@ -0,0 +1,197 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using System.Diagnostics;
+using Aspire.Cli.Backchannel;
+using Aspire.Cli.Interaction;
+using Aspire.Cli.Projects;
+using Aspire.Cli.Utils;
+using Aspire.Hosting;
+
+namespace Aspire.Cli.Commands;
+
+internal sealed class DeployCommand : BaseCommand
+{
+    private readonly ActivitySource _activitySource = new ActivitySource(nameof(DeployCommand));
+    private readonly IDotNetCliRunner _runner;
+    private readonly IInteractionService _interactionService;
+    private readonly IProjectLocator _projectLocator;
+
+    public DeployCommand(IDotNetCliRunner runner, IInteractionService interactionService, IProjectLocator projectLocator)
+        : base("deploy", "Deploy an Aspire app host project to the supported targets.")
+    {
+        ArgumentNullException.ThrowIfNull(runner);
+        ArgumentNullException.ThrowIfNull(interactionService);
+        ArgumentNullException.ThrowIfNull(projectLocator);
+
+        _runner = runner;
+        _interactionService = interactionService;
+        _projectLocator = projectLocator;
+
+        var projectOption = new Option<FileInfo?>("--project");
+        projectOption.Description = "The path to the Aspire app host project file.";
+        Options.Add(projectOption);
+
+        var outputPathOption = new Option<string?>("--output-path", "-o");
```

Comment:
Why does the `deploy` command have an output-path? I wouldn't expect deploy to output anything to disk.

Reply:
Isn't it likely to need a staging location between publish and deploy... Maybe this should be defaulted to a temp directory in case of the deploy option.